### PR TITLE
fix(source): initialize sources of images

### PR DESCRIPTION
**Pull request by mudler**
Should fix the panic in https://github.com/kairos-io/kairos/issues/2203

### CODE REVIEW

File: pkg/config/spec.go [hunk 1]
```diff
@@ -221,6 +222,10 @@ func NewUpgradeSpec(cfg *Config) (*v1.UpgradeSpec, error) {
 	var recLabel, recFs, recMnt string
 	var active, passive, recovery v1.Image
 
+	active.Source = v1.NewEmptySrc()
```
Comment (Itxaka) [inline pkg/config/spec.go:225]: this only affect normal upgrades, which dont have any issues with nil access :D

Fix should og unto `NewUpgradeUkiSpec`

Comment (mudler) [reply pkg/config/spec.go:225]: right indeed.. I've just followed the panic here, did realize afterwards that we were actually skipping the UKI upgrade command and going into normal mode

**Comment by jimmykarily** [comment]
This didn't fix it. We provided the actual fix elsewhere.

**Comment by Itxaka** [comment]
Fixed at https://github.com/kairos-io/kairos-agent/pull/219

**Pull request closed without merging**